Code/Framework/AzCore/AzCore/Debug/Profiler.h

```diff
@@ -318,6 +52,40 @@ namespace AZStd
 
 namespace AZ
 {
+    class ProfileScope
+    {
+    public:
+        static uint32_t GetSystemID(const char* system);
+
+        template<typename... T>
+        static void BeginRegion([[maybe_unused]] const char* system, [[maybe_unused]] char const* eventName, [[maybe_unused]] T const&... args)
+        {
+            // TODO: Verification that the supplied system name corresponds to a known budget
+#if defined(USE_PIX)
+            PIXBeginEvent(PIX_COLOR_INDEX(GetSystemID(system) & 0xff), eventName, args...);
+#endif
+            // TODO: injecting instrumentation for other profilers
```

Comment:
This comment is a bit misleading imho. Long term O3DE will offer two options. By default the AZ::Interface for the profiler will be called to forward the call to a dynamically assigned profiler. Alternatively users can opt to define a path to a profiler through CMake (such as RAD Telemetry) and here the call will go to the directly linked profiler instead. For the latter I'm assume some PAL-like will need to be used so different gems can be used. Having a bunch of different profilers hook in here seems to go against the modular nature.

Reply:
Agreed, this won't be here long term, but the comment is here to indicate we intend on branching the event begin/end out to potentially multiple endpoints. For this change, no scaffolding is yet in place for exactly the mechanism that will do this (which needs to contemplate how dynamic args are handled)

Reply:
Adding a comment to clarify this as a temporary stopgap